geoutils/georaster/raster.py

```diff
@@ -137,40 +177,15 @@ class Raster:
         width
     """
 
-    # This only gets set if a disk-based file is read in.
-    # If the Raster is created with from_array, from_mem etc, this stays as None.
-    filename = None
-    _is_modified: bool | None = None
-    _disk_hash: int | None = None
-
-    # Rasterio-inherited names and types are defined here to get proper type hints.
-    # Maybe these don't have to be hard-coded in the future?
-    _data: np.ndarray | np.ma.masked_array
-    transform: Affine
-    crs: CRS
-    nodata: int | float | None
-    res: tuple[float, float]
-    bounds: rio.coords.BoundingBox
-    height: int
-    width: int
-    shape: tuple[int, int]
-    indexes: list[int]
-    count: int
-    dataset_mask: np.ndarray | None
-    driver: str
-    dtypes: list[str]
-    name: str
-
     def __init__(
         self,
-        filename_or_dataset: str | RasterType | rio.io.DatasetReader | rio.io.MemoryFile,
+        filename_or_dataset: str | RasterType | rio.io.DatasetReader | rio.io.MemoryFile | dict[str, Any],
         bands: None | int | list[int] = None,
         load_data: bool = True,
         downsample: AnyNumber = 1,
         masked: bool = True,
```

Comment:
This is somewhat unrelated to this PR, but should we not enforce that self.data is a masked array?

Reply:
It is now enforced in d690b1d